### ISSUE TITLE
Proposal: Improve SVG APIs

### ISSUE THREAD

**Issue by HppZ** [Microsoft/microsoft-ui-xaml#507]
# Proposal: Improve SVG APIs

## Summary

SvgImageSource needs dynamically change colors. 
as discussed here:
https://github.com/MicrosoftDocs/winrt-api/issues/375

**Comment by verelpode** [comment]
I guess that @HppZ would like to use a SVG file in a manner similar to setting [Path.Data](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.shapes.path.data) along with setting [Path.Fill](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.shapes.shape.fill) to the desired brush/color.   In other words, the ability to convert an SVG file to a [Geometry](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.media.geometry) that can be assigned to [Path.Data](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.shapes.path.data), thereby allowing the brush/color to be dynamically changed at any time by changing [Path.Fill](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.shapes.shape.fill).

For example, consider an SVG file that should be used as an icon for a button.  Thus the icon/symbol should change color depending on the state of the button (pressed, disabled, hover, etc).

I agree that it would be great to be able to use SVG files as icons for buttons or as symbols for other parts of the GUI.  Thus both "colored" and "colorless" SVG files are quite useful in different situations.

**Comment by HppZ** [comment]
Yes, I would like to use a SVG in a Button style so that I don't need three images (normal pressed hover).
and no more scaled assets needed.

**Comment by jesbis** [comment]
@HppZ, @verelpode do you have any thoughts on which of these would be most useful:

1. Ability to convert an svg document to a Xaml Geometry
2. Ability to convert an svg document to a Win2D or Windows.UI.Composition geometry/shape
3. Ability to apply Xaml styles to svg elements in an SvgImageSource (e.g. similar to styling svg elements with CSS classes)

?

**Comment by verelpode** [comment]
Considering that we both thought of using SVG files/resources as icons for buttons or as other symbols in the GUI, and considering that buttons are normally composed of Xaml elements, then I think the most useful ability is to convert SVG to Xaml Geometry (not CompositionGeometry) and assign it to `Path.Data`.  Importantly, this also makes it compatible with `Windows.UI.Xaml.Controls.PathIcon.Data`, whereas CompositionGeometry cannot be assigned to `PathIcon.Data`.

Re the third option, that may be impressively advanced but overkill; too much work.  The simpler solution of converting an SVG to Xaml Geometry would make it easier to use an SVG as an symbol for use with `PathIcon.Data` or `Path.Data`.

**Comment by jesbis** [comment]
Thanks for the suggestion and details! I agree this would be a useful feature given how common SVG is.
We won't be able to get to this in the current release but we'll keep this issue in the backlog for the future.

**Comment by ArchieCoder** [comment]
I was searching how to do this until I find out this thread :(

For reference iOS/Android, we can do it. Swift way: https://stackoverflow.com/questions/19274789/how-can-i-change-image-tintcolor-in-ios-and-watchkit

**Comment by hupo376787** [comment]
For svg style, we can use [Glyphter](https://glyphter.com/) to convert svg to font file, and do whatever we want. We can't rely on MS, we have to find tools online.

**Comment by agneszitte** [comment]
Any future plan for this proposal please?

**Comment by Marv51** [comment]
For my app, it would already be helpful if SvgImageSource could optionally behave like a BitmapIcon. 

When a property `RenderMonochrome` is set on the SvgImageSource, the svg would be rendered monochrome in the foreground color.
